src/python/pants/backend/python/rules/pex_tools/pex_tool.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os
+from abc import ABC
+from typing import ClassVar, Iterable, List
+
+import pkg_resources
+
+from pants.engine.fs import FileContent, FilesContent, InputFilesContent
+
+_PACKAGE: List[str] = __name__.split(".")[:-1]
+
+
+class PexTool(ABC):
+    _entry_point: ClassVar[str]
+    _module_names: ClassVar[Iterable[str]]
+
+    @classmethod
+    def entry_point(cls):
+        return f"{'.'.join(_PACKAGE)}.{cls._entry_point}"
+
+    @classmethod
+    def files(cls) -> InputFilesContent:
+        return InputFilesContent(
+            FilesContent(
+                FileContent(
+                    path=f"{os.path.join(*_PACKAGE, module_name)}.py",
+                    content=pkg_resources.resource_string(__name__, f"{module_name}.py"),
+                )
+                for module_name in cls._module_names
+            )
+        )
```

Comment:
I think it'd be better to return `FilesContent` and have the call site wrap it in `InputFilesContent`. Per this comment, we don't even want `InputFilesContent` to exist:

https://github.com/pantsbuild/pants/blob/766ce1d0127a3c59afb6d8cca533616fa427e57d/src/python/pants/engine/fs.py#L43-L48

--

If you stick with this return type, you can remove the intermediate `FilesContent`. The constructor for `InputFilesContent` expects `Input[FileContent]`, so the generator will do.